mypy/checker.py

```diff
@@ -2829,6 +2839,19 @@ def infer_partial_type(self, name: Var, lvalue: Lvalue, init_type: Type) -> bool
         self.partial_types[-1].map[name] = lvalue
         return True
 
+    def is_valid_ordereddict_partial_value_type(self, t: Instance) -> bool:
+        if len(t.args) == 0:
+            return True
+        if len(t.args) == 1:
+            arg = get_proper_type(t.args[0])
+            # TODO: This is too permissive -- we only allow TypeVarType since
+            #       they leak in cases like defaultdict(list) due to a bug.
+            #       This can result in incorrect types being inferred, but only
+            #       in rare cases.
+            if isinstance(arg, (TypeVarType, UninhabitedType)):
```

Comment:
Should this also include `NoneType`? Otherwise it may not work if `strict_optional=False` (see the check for key type). Add a test for this.

Reply:
I'll add a test.